Subject: [PATCH] git-instaweb: set better defaults

Enable blame, pathinfo, better diff and set charset to utf-8.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 git-instaweb.sh | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/git-instaweb.sh b/git-instaweb.sh
index b9366a56..5ac32127 100755
--- a/git-instaweb.sh
+++ b/git-instaweb.sh
@@ -257,6 +257,7 @@ server.modules = ( "mod_setenv", "mod_cgi" )
 server.indexfiles = ( "gitweb.cgi" )
 server.pid-file = "$fqgitdir/pid"
 server.errorlog = "$fqgitdir/gitweb/$httpd_only/error.log"
+server.force-lowercase-filenames = "disable"
 
 # to enable, add "mod_access", "mod_accesslog" to server.modules
 # variable above and uncomment this
@@ -597,8 +598,20 @@ gitweb_conf() {
 our \$projectroot = "$(dirname "$fqgitdir")";
 our \$git_temp = "$fqgitdir/gitweb/tmp";
 our \$projects_list = \$projectroot;
+our \$default_text_plain_charset = 'utf-8';
+our \$my_uri = "/gitweb.cgi";
+our \$home_link = "/gitweb.cgi";
+our @stylesheets = ("/static/gitweb.css");
+our \$logo = "/static/git-logo.png";
+our \$favicon = "/static/git-favicon.png";
+our \$javascript = "/static/gitweb.js";
+our @diff_opts = ('-B', '-C');
 
 \$feature{'remote_heads'}{'default'} = [1];
+\$feature{'blame'}{'default'} = [1];
+\$feature{'pathinfo'}{'default'} = [1];
+\$feature{'blame_incremental'}{'default'} = [1];
+\$feature{'timed'}{'default'} = [1];
 EOF
 	hlbin="$(command -v highlight 2>/dev/null)"
 	if [ -x "$hlbin" ] && "$hlbin" --version | grep -q 'Andre Simon'; then
---
